# Assignment 3 of HW3
## Finish z-test lab

### Null Hypothesis:
    The new bus route will result in a commute of the same or longer duration as the old route. This can be represented as T(old) <= T(new). This will be tested against a significance level of α = 0.05, i.e. 2σ from the old mean.

### Alternative Hypothesis;
    The new bus route will result in a shorter commute than the old route. This can be represented as T(new) < T(old)

In [1]:
from __future__  import print_function, division
import numpy as np

In [2]:
!curl -O https://raw.githubusercontent.com/fedhere/PUI2016_fb55/master/Lab3_fb55/times.txt

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
107  1179  107  1179    0     0   9275      0 --:--:-- --:--:-- --:--:--  383k


In [3]:
newRoute = np.loadtxt('./times.txt', dtype=float)
newMean = newRoute.mean()
print (newMean)
newRoute.size

34.4661616883


100

#### The duration of the new commute was measured 100 times ('N') and the resulting sample mean is 34.466 
#### The mean and standard deviation for the old bus route were given as μ = 36 and σ = 6

In [4]:
oldMean = 36
standev = 6
size = 100

#### The  z-score is calculated using the following formula:

In [5]:
zScore = ((oldMean - newMean) / ((standev)/(np.sqrt(size))))

In [6]:
zScore

2.5563971861666701

The z-score comes out to 2.556, which is greater than 2. Therefore, as this is more than two standard deviations away from the old mean, i.e. not within the specified α of 5%, we can reject the Null Hypothesis.